src/scenes/challengeMode.js

```diff
@@ -1,15 +1,75 @@
 import Scene from "../objects/scene.js";
+import ConstantGate from "../objects/constantGate.js";
+import AndGate from "../objects/andGate.js";
 
 /**
  * This scene will manage the game's challenge mode.
  */
 export default class ChallengeMode extends Scene {
     constructor() {
         super();
+        this.bufferImg = null;
+        this.notImg = null;
+        this.andImg = null;
+        this.orImg = null;
+        this.xorImg = null;
+        this.nandImg = null;
+        this.norImg = null;
+        this.xnorImg = null;
+    	this.constant1 = null;
+    	this.constant2 = null;
+    	this.gate = null;
     }
 
-    setup() {}
+    preload()
+    {
+    	let promises = [
```

Comment:
Will this be made redundant by `GateFactory.preload()` in #51 ?

Reply:
Yes it will, it is basically all thrown into the gate factory preloader, that mess is for demonstration purposes.